### PR TITLE
Feat 2.1.x/ab#65456 sorting fields widget option

### PR DESCRIPTION
**Pull request by NathanHGit**
# Description

Added a sort select to summary cards and grid widget which is configurable from the corresponding widget options. Created a new component to manage the list of sort fields.

## Ticket

https://dev.azure.com/WHOHQ/EMSSAFE/_workitems/edit/65456

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

- I created both summary cards and grid widgets
- I selected the "Allow sorting" button in the widget display panel from widget options
- I added at least one field corresponding with the current layout in the list of the sorting panel
- I tried to sort my widgets (the two displays of summary cards and the grid widget)

## Sreenshots

image

image

image

image

# Checklist:

( * == Mandatory ) 

- [X] * My code follows the style guidelines of this project
- [X] * Linting does not generate new warnings
- [X] * I have performed a self-review of my own code
- [X] * I have commented my code, particularly in hard-to-understand areas
- [X] * I have put JSDoc comment in all required places
- [X] * My changes generate no new warnings
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have included screenshots describing my changes if relevant
- [X] * I have selected labels in the Pull Request, according to the changes with code brings
- [ ] I have made corresponding changes to the documentation ( if required )
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/safe/src/lib/components/widgets/common/filtering-settings/filtering-settings.component.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  /** Displayed columns of table */
+  public displayedColumnsApps = ['field', 'order', 'label', 'actions'];
+
+  public items = [
+    { field: 'name', order: 'asc', label: 'label1' },
+    { field: 'height', order: 'desc', label: 'label2' },
+  ];
+
+  public rows: FormArray = this.formBuilder.array([]);
+  public formGroup: FormGroup = this.formBuilder.group({ filters: this.rows });
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/widgets/common/filtering-settings/filtering-settings.component.ts]: @NathanHGit there you need to use an input formArray, and build the formArray from inside the summary-card-settings, so you can store it

File: libs/safe/src/lib/components/widgets/summary-card/summary-card.component.html [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <!-- Header -->
   <kendo-tilelayout-item-header
     *ngIf="header"
-    class="widget-header border-b"
+    class="widget-header border-b h-[45px]"
```
Comment (matheus-relief) [inline libs/safe/src/lib/components/widgets/summary-card/summary-card.component.html:5]: I also moved the sort selection to the top, in the header beside the new search bar, and had to increase the size a bit to make it fit well, not sure you'll like it

image

Other than that, lgtm!

**Review by Maximo-reliefapps — APPROVED** [review]
Merged latest version of 2.1.x.
Fixed a bug duplicating sort options (originated from the merge)
Fixed wrong style for the summary cards sort.

The sort feature was tested in grids and summary cards, working as expected

File: libs/safe/src/lib/components/ui/core-grid/grid/grid.component.html [hunk 3]
```diff
@@ -97,6 +97,20 @@
...
+    <kendo-combobox
+      class="w-fit"
+      *ngIf="
+        !!widget?.settings.widgetDisplay?.sortable &&
+        data.total &&
+        sortFields.length > 0
+      "
+      [data]="sortFields"
+      textField="label"
+      valueField="label"
```
Comment (AntoineRelief) [inline libs/safe/src/lib/components/ui/core-grid/grid/grid.component.html]: @NathanHGit 
⚠️ that does not sound right, to have valueField="label"

**Comment by AntoineRelief** [comment]
Capture d’écran 2023-08-21 à 11 30 08

@NathanHGit @matheus-relief 
ℹ️  please also make sure the UI is not confusing
there, we have widget display to describe sorting, which is confusing

**Pull request merged**